### PR TITLE
Add PrintLastLog, UsePrivilegeSeparation and Compression params

### PR DESCRIPTION
**Pull request by Mears148**
This will allow for those trying to harden their systems to comply with security standards.

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -482,6 +485,7 @@
     it { should contain_file('sshd_config').with_content(/^PermitRootLogin no$/) }
     it { should contain_file('sshd_config').with_content(/^ChallengeResponseAuthentication no$/) }
     it { should contain_file('sshd_config').with_content(/^PrintMotd no$/) }
+    it { should contain_file('sshd_config').with_content(/^PrintLastLog no$/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: There should be lines like this for the other two parameters

**Comment by ghoneycutt** [comment]
Wow! Thanks for the very complete PR!

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (Mears148) [inline spec/classes/init_spec.rb]: Sorry about that...I'm just learning. I will add those two lines. Thanks for the feedback.

**Comment by Mears148** [comment]
No problem, thanks for the encouragement...I am working through some of
these issues with the builds, and then I will do a squash so it isn't so
messy.

**Comment by ghoneycutt** [comment]
Thanks! You can test this locally by running `bundle install && bundle exec rake spec_clean spec_prep` then `bundle exec rake validate lint spec_standalone`

**Comment by Mears148** [comment]
Preciate the assist with the local test idea. This should be better now. Thanks!

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -960,6 +966,26 @@
       end
     end
   end
+	
+  describe 'sshd_config_compression param' do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: We need a similar section for `sshd_config_print_lastlog`

**Comment by ghoneycutt** [comment]
@Mears148 I appreciate all your work! One param is missing a testing stanza, other than that, looks great!

For anyone else following: if you look at this PR, there are a lot of whitespace changes. To view only real changes, add `?w=1` to the URL.

https://github.com/ghoneycutt/puppet-module-ssh/pull/225/files?w=1

File: spec/classes/init_spec.rb (hunk 2 above)
Comment (Mears148) [inline spec/classes/init_spec.rb]: I added in the section...let me know if I am missing anything else. Thanks for all the helpful feedback and tips!

Comment (Mears148) [inline spec/classes/init_spec.rb]: Were there any other changes you needed me to make? Thanks.

File: README.md [hunk 3]
```diff
@@ -259,6 +259,14 @@ Valid values are 'yes' and 'no'.
 
 - *Default*: undef
 
+sshd_config_compression
+---------------------------------
+Compression option in sshd_config.
+Specifies whether compression is allowed in an SSH connection prior to authentication.
```
Comment (ghoneycutt) [inline README.md]: `If specified, valid values are 'yes', 'no' and 'delayed'.`

File: README.md [hunk 4]
```diff
@@ -298,6 +306,13 @@ PrintMotd option in sshd_config.
 
 - *Default*: 'yes'
 
+sshd_config_print_lastlog
+----------------------
+PrintLastLog option in sshd_config.
+Verify SSH provides users with feedback on when account accesses last occurred.
```
Comment (ghoneycutt) [inline README.md]: `If specified, valid values are 'yes' and 'no'.`

File: README.md [hunk 5]
```diff
@@ -470,6 +485,13 @@ On Solaris the default is to not add this parameter to the configuration file.
 
 - *Default*: undef
 
+sshd_config_use_privilege_separation
+----------------------
+UsePrivilegeSeparation in sshd_config.
+Causes the SSH process to drop root privileges when not needed.
```
Comment (ghoneycutt) [inline README.md]: `if specified, valid values are 'yes', 'no' and 'sandbox'.`

**Comment by ghoneycutt** [comment]
Should document what the valid values are. Great you are already validating for them.

**Comment by ghoneycutt** [comment]
The docs are the last thing. Could you rebase into one commit and I'll get this released.

**Comment by Mears148** [comment]
Done...thanks again!

**Comment by ghoneycutt** [comment]
Thank you!

Released in v3.52.0

**Comment by Mears148** [comment]
Glad I could contribute! Thanks for all the tips!

**Pull request merged**